fetch.bs

```diff
@@ -8485,6 +8618,92 @@ with a <var>promise</var>, <var>request</var>, <var>responseObject</var>, and an
 </div>
 
 
+<h3 id=fetch-later-method>FetchLater method</h3>
+
+<pre class=idl>
+
+dictionary DeferredRequestInit : RequestInit {
+  DOMHighResTimeStamp backgroundTimeout;
+};
+
+interface FetchLaterResult {
+  readonly attribute boolean activated;
+};
+
+partial interface mixin WindowOrWorkerGlobalScope {
+  [NewObject] FetchLaterResult fetchLater(RequestInfo input, optional DeferredRequestInit init = {});
+};
+</pre>
+
+<p>A {{FetchLaterResult}} has an associated <a>deferred fetch record</a>
+<dfn for=FetchLaterResult>deferred record</dfn>.
+
+<div algorithm>
+<p>The <dfn attribute for=FetchLaterResult><code>activated</code></dfn> getter steps are to return
+true if <a>this</a>'s <a for=FetchLaterResult>deferred record</a>'s
+<a for="deferred fetch record">invoke state</a> is "<code>activated</code>"; Otherwise false.
+</div>
+
+<div algorithm="dom-fetch-later">
+<p>The
+<dfn id=dom-global-fetch-later method for=WindowOrWorkerGlobalScope><code>fetchLater(<var>input</var>, <var>init</var>)</code></dfn>
+method steps are:
+
+<ol>
+ <li>
+  <p>If the user-agent has determined that deferred fetching is not allowed in this context, then
+  throw a {{NotAllowedError}}.
+
+  <p class=note>User-agents can, for example, decide that deferred fetching is not allowed for
+  subframes or gated behind some permission.
+ </li>
+
+ <li><p>Let <var>requestObject</var> be the result of invoking the initial value of {{Request}} as
+ constructor with <var>input</var> and <var>init</var> as arguments. This may throw an exception.
+
+ <li><p>If <var>requestObject</var>'s <a for=Request>signal</a> is <a for=AbortSignal>aborted</a>,
+ then throw <a for=Request>signal</a>'s <a for=AbortSignal>abort reason</a>.
+
+ <li><p>Let <var>request</var> be <var>requestObject</var>'s <a for=Request>request</a>.
+
+ <li><p>If <var>request</var>'s <a for=request>URL</a>'s <a for=url>scheme</a> is not an
+ <a>HTTP(S) scheme</a> then throw a {{TypeError}}.
```

Comment:
```
If request’s URL’s scheme is not an HTTP(S) scheme then throw a TypeError.
```

Should this be HTTPS-only? (although there's a potentially trustworthy check below)

According to our previous [discussion] and https://github.com/WICG/pending-beacon/issues/27, our new API should enforce security requests.

Reply:
@noamr This one is not addressed.

Reply:
Look at the next line, we allow either https or localhost ("potentially trustworthy url")

Reply:
Thanks. Resolved